File: src/main/javascript/binarysearch/Omitted.js
class Solution {
    solve(nums) {
        let isZeroExist = false;
        let zeroCount = 0;
        let max = 1;

        for (let i = 0, len = nums.length; i < len; i++) {
            if (nums[i] === 0) {
                isZeroExist = true;
                zeroCount++;
                continue;
            }
            max *= nums[i];
        }

        let result = [];
        for (let i = 0, len = nums.length; i < len; i++) {
            if (isZeroExist) {
                if (zeroCount > 1) {
                    result.push(0);
                } else {
                    if (nums[i] === 0) {
                        result.push(max);
                    } else {
                        result.push(0);
                    }
                }
            } else {
                result.push(Math.floor(max / nums[i]));
            }
        }
        return result;
    }
}

solution = new Solution();
console.log(solution.solve([0,0]));

File: src/main/golang/binarysearch/Omitted.go
package binarysearch

func omitted(nums []int) []int {
	isZeroExist := false
	zeroCount := 0
	max := 1

	for i, len := 0, len(nums); i < len; i++ {
		if nums[i] == 0 {
			isZeroExist = true
			zeroCount++
			continue
		}
		max *= nums[i]
	}

	result := make([]int, len(nums))
	for i, len := 0, len(result); i < len; i++ {
		if isZeroExist {
			if zeroCount > 1 {
				result[i] = 0
			} else {
				if nums[i] == 0 {
					result[i] = max
				} else {
					result[i] = 0
				}
			}
		} else {
			result[i] = max / nums[i]
		}
	}
	return result
}
